### PR TITLE
Version compatible to v12.0

### PR DESCRIPTION
**Pull request by linawolf**
Features many new examples, automatic tests for coding guidelines and is source for the most recent Documentation in TYPO3 explained.

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -1,3 +1,2 @@
-/.editorconfig export-ignore
-/.gitattributes export-ignore
-/.gitignore export-ignore
+/.* export-ignore
```
Comment (oliverklee) [inline .gitattributes:1]: It looks like this change overwrites the recent changes to this file.

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,4 @@
 /*.idea
+/.Build
 /composer.lock
-/public
-/vendor
+*GENERATED*
```
Comment (oliverklee) [inline .gitignore:4]: Ditto.

File: Build/.gitignore [hunk 3]
```diff
@@ -0,0 +1 @@
+testing-docker/.env
```
Comment (oliverklee) [inline Build/.gitignore:1]: Why a second `.gitignore`?

File: .gitattributes (hunk 1 above)
Comment (linawolf) [reply .gitattributes:1]: This change incooportes your change. as /.* includes all the above. We can also list all the files I just saw that they are doing it like that in https://github.com/TYPO3/styleguide/blob/main/.gitattributes

File: .gitignore (hunk 2 above)
Comment (linawolf) [reply .gitignore:4]: public and vendor are not needed anymore in the .gitignore as all sources are installed to /.Build. See changes in the composer.json

File: Build/.gitignore (hunk 3 above)
Comment (linawolf) [reply Build/.gitignore]: moved them into one

File: .gitattributes (hunk 1 above)
Comment (oliverklee) [reply .gitattributes:1]: Ack

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    tags:
+      - "**"
+  workflow_dispatch:
+
+jobs:
+  publish:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
```
Comment (oliverklee) [inline .github/workflows/publish.yml:14]: Checkout now has been released in version 3.

File: Classes/Controller/AbstractController.php [hunk 5]
```diff
@@ -16,55 +17,78 @@
...
-        $locallangKey = sprintf('error.%s.%s', $this->request->getControllerName(), $this->actionMethodName);
+        $locallangKey = sprintf(
+            'error.%s.%s',
+            $this->request->getControllerName(),
+            $this->actionMethodName
+        );
         return $this->translate($locallangKey, $defaultFlashMessage);
     }
 
+    protected function hasBlogAdminAccess(): boolean
```
Comment (oliverklee) [inline Classes/Controller/AbstractController.php:45]: `boolean` -> `bool`

File: .gitignore (hunk 2 above)
Comment (oliverklee) [reply .gitignore:4]: Ack.

**Comment by DavidBruchmann** [comment]
There are some inconsistencies in syntax but some will be replaced anyway and others can be adjusted later.
For the moment it's important to get this branch running.

**Pull request merged**